### PR TITLE
fix work hash confilict and compare mainfests

### PR DESCRIPTION
**Pull request by zhiweiyin318**

### CODE REVIEW

File: utils/work/v1/workapplier/workapplier.go [hunk 1]
```diff
@@ -121,7 +126,33 @@ func manifestsEqual(new, old []workapiv1.Manifest) bool {
...
+}
+
+func manifestUnstructured(manifest workapiv1.Manifest) *unstructured.Unstructured {
+	newManifest := &unstructured.Unstructured{}
+	if err := newManifest.UnmarshalJSON(manifest.Raw); err != nil {
+		klog.Errorf("failed to unmarshal new manifest. err:%v", err)
+		return newManifest
+	}
+
+	unstructured.RemoveNestedField(newManifest.Object, "metadata", "creationTimestamp")
```
Comment (qiujian16) [inline utils/work/v1/workapplier/workapplier.go]: need to add some comments here

File: utils/work/v1/workapplier/workapplier.go [hunk 2]
```diff
@@ -121,7 +126,33 @@ func manifestsEqual(new, old []workapiv1.Manifest) bool {
...
+	unstructured.RemoveNestedField(newManifest.Object, "spec", "template", "metadata", "creationTimestamp")
+
+	return newManifest
+}
+
+// mapExisted is used to check if required is included in existing.
+func mapExisted(required, existing map[string]string) bool {
```
Comment (qiujian16) [inline utils/work/v1/workapplier/workapplier.go]: I think it is more like `shouldUpdateMap` ?

File: utils/work/v1/workapplier/workapplier.go (hunk 1 above)
Comment (zhiweiyin318) [reply utils/work/v1/workapplier/workapplier.go]: added

File: utils/work/v1/workapplier/workapplier.go (hunk 2 above)
Comment (zhiweiyin318) [reply utils/work/v1/workapplier/workapplier.go]: yes, changed the func name.

**Comment by zhiweiyin318** [comment]
@qiujian16   please review again.  addressed the comments we discussed.

**Pull request merged**